src/Microsoft.Diagnostics.Monitoring.EventPipe/Triggers/Pipelines/TraceEventTriggerPipeline.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Tracing;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.EventPipe.Triggers.Pipelines
+{
+    /// <summary>
+    /// A pipeline that detects a condition (as specified by the trigger) within the event stream
+    /// of the specified event source. The callback is invoked for each instance of the detected condition.
+    /// </summary>
+    internal sealed class TraceEventTriggerPipeline : Pipeline
```

Comment:
I was initially going to suggest that this isn't really a Pipeline so much as an implementation detail, but it sounds like you want to be able to feed TraceEventSource directly into the TriggerPipeline.